tensor_comprehensions/library/layers.yaml

```diff
@@ -0,0 +1,247 @@
+      O1(b, n) = O1(b, n) + B1(n)
+    }
+
+- name: relu
+  lang: |
+    def relu(float(B,M) I) -> (O1){
+      O1(b, m) = fmax(I(b, m), 0)
+    }
+
+- name: fcrelu
+  lang: |
+    def fcrelu(float(B,M) I, float(N,M) W1, float(N) B1) -> (O1){
+      O1(b, n) +=! I(b, m) * W1(n, m)
+      O1(b, n) = O1(b, n) + B1(n)
+      O1(b, n) = fmax(O1(b, n), 0)
+    }
+
+- name: cast
+  lang: |
+    def cast(float(M,N) A) -> (int32(M,N) O1) {{
+      O1(m, n) = int32(A(m, n) + {four})
+    }}
+
+- name: concat
+  lang: |
+    def concat(float(M, N) A, float(M, N) B) -> (O1) {
+      O1(n, i, m) = i == 0 ? A(m, n) : B(m, n) where i in 0:2
+    }
+
+- name: convolution
+  lang: |
+    def convolution(float(N,C,H,W) I, float(M,C,KH,KW) W1, float(M) B) -> (O) {
+      O(n, m, h, w) +=! I(n, c, h + kh, w + kw) * W1(m, c, kh, kw)
+      O(n, m, h, w) = O(n, m, h, w) + B(m)
+    }
+
+- name: convolution_strided
+  lang: |
+    def convolution_strided(float(N,C,H,W) I, float(M,C,KH,KW) W1, float(M) B) -> (O) {{
+      O(n, m, h, w) +=! I(n, c, {sh} * h + kh, {sw} * w + kw) * W1(m, c, kh, kw)
+      O(n, m, h, w) = O(n, m, h, w) + B(m)
+    }}
+  grad: |
+    def convolution_strided_grad(float(N,C,H,W) I, float(M,C,KH,KW) W1, float(N,M,H,W) O_grad)
+    -> (I_grad, W1_grad) {{
+      I_grad(n, c, h, w) +=! O_grad(n, m, {sh} * h - kh, {sw} * w - kw) * W1(m, c, kh, kw)
+      W1_grad(m, c, kh, kw) +=! O_grad(n, m, {sh} * h - kh, {sw} * w - kw) * I(n, c, h, w)
+    }}
+
+- name: group_convolution
+  lang: |
+    def group_convolution(float(N,G,C,H,W) I, float(G,F,C,KH,KW) W1, float(G,F) B) -> (O)
+    {
+      O(n, g, f, h, w) +=! I(n, g, c, h + kh, w + kw) * W1(g, f, c, kh, kw)
+      O(n, g, f, h, w) = O(n, g, f, h, w) + B(g, f)
+    }
+
+- name: group_convolution_strided
+  lang: |
+    def group_convolution_strided(float(N,G,C,H,W) I, float(G,F,C,KH,KW) W1, float(G,F) B) -> (O)
+    {{
+      O(n, g, f, h, w) +=! I(n, g, c, {sh} * h + kh, {sw} * w + kw) * W1(g, f, c, kh, kw)
+      O(n, g, f, h, w) = O(n, g, f, h, w) + B(g, f)
+    }}
+
+- name: copy2D
+  lang: |
+    def copy2D(float(M, N) I) -> (O) {
+      O(i, j) = I(i, j)
+    }
+
+- name: copy
+  lang: |
+    def copy(float({dimParams}) I) -> (O) {{
+      O({dimIndices}) = I({dimIndices})
+    }}
+
+- name: cosine
+  lang: |
+    def cosine(float(M) I) -> (O) {
+      O(i) = cos(I(i))
+    }
+
+- name: cosine_similarity
+  lang: |
+    def cosine_similarity(float(M, N) I1, float(M, N) I2) -> (O, sumI1, sumI2) {{
+      sumI1(m) +=! I1(m, n) * I1(m, n)
+      sumI2(m) +=! I2(m, n) * I2(m, n)
+      O(m) +=! (I1(m, n) * I2(m, n)) / fmax(rsqrt(sumI1(m)) * sqrt(sumI2(m)), {eps})
+    }}
+
+- name: add
+  lang: |
+    def add(float(N) A, float(N) B) -> (output) {
+      output(i) = A(i) + B(i) + 1
+    }
+
+- name: abs
+  lang: |
+    def abs(float(M, N) A) -> (O1) {
+      O1(m, n) = fabs(A(m, n))
+    }
+
+- name: layernorm
+  lang: |
+    def layernorm(float(T, B, C) I) -> (O, mean, centered, var)
```

Comment:
this is one of the few that didn't exist in https://github.com/facebookresearch/TensorComprehensions/tree/master/include/tc/library
please consider opening a task so that we have parity between C++ and Python